synapse/module_api/__init__.py

```diff
@@ -1520,6 +1521,26 @@ async def get_monthly_active_users_by_service(
             start_timestamp, end_timestamp
         )
 
+    async def get_room_alias(self, room_id: RoomID) -> Optional[RoomAlias]:
+        """
+        Get the room alias associated with a room ID.
+
+        Added in Synapse v1.82.0.
+
+        Args:
+            room_id: The Room ID to find the alias of.
+
+        Returns:
+            The optional room alias
+        """
+        from synapse.api.constants import EventTypes
+
+        alias_event = await self._storage_controllers.state.get_current_state_event(
+            room_id, EventTypes.CanonicalAlias, ""
+        )
+        if alias_event:
+            return alias_event.content.get("alias")
```

Comment:
This value is going to be an `Optional[str]`, not an `Optional[RoomAlias]`; you'll need to call `RoomAlias.from_string()`  on this value before returning.

Reply:
Done.